protocol/contracts/beanstalk/barn/UnripeFacet.sol

```diff
@@ -141,13 +136,11 @@ contract UnripeFacet is ReentrancyGuard {
         view
         returns (uint256 redeem)
     {
-        require(isUnripe(unripeToken), "not vesting");
-        uint256 sharesBeingRedeemed = getRecapPaidPercentAmount(amount);
-        redeem = _getUnderlying(unripeToken, sharesBeingRedeemed, supply);
+        redeem = LibChop._getPenalizedUnderlying(unripeToken, amount, supply);
     }
 
     function isUnripe(address unripeToken) public view returns (bool unripe) {
-        unripe = s.u[unripeToken].underlyingToken != address(0);
+        unripe = LibChop.isUnripe(unripeToken);
```

Comment:
can we make this external

Reply:
As i recall @publiuss  mentioned that i use LibChop in all public functions in UnripeFacet so that they can be called off-chain. Changing this to external modifies that. What would be the reasoning behind this? (Fixed in: https://github.com/BeanstalkFarms/Beanstalk/pull/460/commits/64c22d5e6930522fdeb7c0fb94501d3fb07366a5)

Reply:
the major difference between calling a function public and external is that a public function can be called internally in a contract, whereas an external function cannot. If a function is only called externally, it's better to denote it as external for 1) gas savings and 2) tells developers that the function is only called externally. In other words, if I saw a public function, i'd be looking within the code base to see where the function is called.